pkg/solana/chainwriter/transform_registry.go

```diff
@@ -2,49 +2,105 @@ package chainwriter
+	ReportContext [2][32]byte
+	Report        []byte
+	Info          ccipocr3.ExecuteReportInfo
+	TokenIndexes  []byte
+}
+
+type ExtraDataDecoded struct {
+	// ExtraArgsDecoded contain message specific extra args.
+	ExtraArgsDecoded map[string]any
+	// DestExecDataDecoded contain token transfer specific extra args.
+	DestExecDataDecoded []map[string]any
 }
 
-func FindTransform(id string) (func(context.Context, any, solana.AccountMetaSlice, map[string]map[string][]*solana.AccountMeta) (any, solana.AccountMetaSlice, error), error) {
+type SVMExecCallArgs struct {
+	ReportContext [2][32]byte                `mapstructure:"ReportContext"`
+	Report        []byte                     `mapstructure:"Report"`
+	Info          ccipocr3.ExecuteReportInfo `mapstructure:"Info"`
+	ExtraData     ExtraDataDecoded           `mapstructure:"ExtraData"`
+}
+
+type SVMCommitCallArgs struct {
+	ReportContext [2][32]byte               `mapstructure:"ReportContext"`
+	Report        []byte                    `mapstructure:"Report"`
+	Rs            [][32]byte                `mapstructure:"Rs"`
+	Ss            [][32]byte                `mapstructure:"Ss"`
+	RawVs         [32]byte                  `mapstructure:"RawVs"`
+	Info          ccipocr3.CommitReportInfo `mapstructure:"Info"`
+}
+
+// TODO: replace with actual value from CCIP on-chain
+const StaticCuOverhead uint32 = 120000
+
+func FindTransform(id string) (func(context.Context, any, solana.AccountMetaSlice, map[string]map[string][]*solana.AccountMeta) (any, solana.AccountMetaSlice, []txmutils.SetTxConfig, error), error) {
 	switch id {
 	case "CCIPExecute":
-		return CCIPExecuteArgsTransform, nil
+		return CCIPExecuteTransform, nil
 	case "CCIPCommit":
-		return CCIPCommitAccountTransform, nil
+		return CCIPCommitTransform, nil
 	default:
 		return nil, fmt.Errorf("transform not found")
 	}
 }
 
 // This Transform function looks up the token pool addresses in the accounts slice and augments the args
 // with the indexes of the token pool addresses in the accounts slice.
-func CCIPExecuteArgsTransform(ctx context.Context, args any, accounts solana.AccountMetaSlice, tableMap map[string]map[string][]*solana.AccountMeta) (any, solana.AccountMetaSlice, error) {
-	var argsTransformed ReportPostTransform
-	err := mapstructure.Decode(args, &argsTransformed)
+func CCIPExecuteTransform(ctx context.Context, args any, accounts solana.AccountMetaSlice, tableMap map[string]map[string][]*solana.AccountMeta) (any, solana.AccountMetaSlice, []txmutils.SetTxConfig, error) {
+	var argsDecoded SVMExecCallArgs
+	err := mapstructure.Decode(args, &argsDecoded)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, []txmutils.SetTxConfig{}, err
+	}
+
+	argsTransformed := ReportPostTransform{
+		ReportContext: argsDecoded.ReportContext,
+		Report:        argsDecoded.Report,
+		Info:          argsDecoded.Info,
+	}
+
+	if len(argsTransformed.Info.AbstractReports) != 1 || len(argsTransformed.Info.AbstractReports[0].Messages) != 1 {
```

Comment:
We didn't want to enforce this right? In case we decide to support multiple messages in the future?

Reply:
yep good catch forgot to remove this